lib/pure/math.nim

```diff
@@ -880,6 +880,37 @@ func floorMod*[T: SomeNumber](x, y: T): T =
   result = x mod y
   if (result > 0 and y < 0) or (result < 0 and y > 0): result += y
 
+func euclDiv*[T: SomeInteger](x, y: T): T {.since: (1, 5, 1).} =
+  ## Returns euclidean division of `x` by `y`.
+  runnableExamples:
+    assert euclDiv(13, 3) == 4
+    assert euclDiv(-13, 3) == -5
+    assert euclDiv(13, -3) == -4
+    assert euclDiv(-13, -3) == 5
+  result = x div y
+  let r = x mod y
+  if r < 0:
+    if y > 0:
+      dec result
+    else:
+      inc result
+
+func euclMod*[T: SomeNumber](x, y: T): T {.since: (1, 5, 1).} =
+  ## Returns euclidean modulo of `x` by `y`.
+  ## `euclMod(x, y)` is non-negative.
+  runnableExamples:
+    assert euclMod(13, 3) == 1
+    assert euclMod(-13, 3) == 2
+    assert euclMod(13, -3) == 1
+    assert euclMod(-13, -3) == 2
+  result = x mod y
+  if result < 0:
+    # increase by abs(y)
+    if y >= 0:
```

Comment:
ok.
In future work I'm wondering if this would be better:

`if result < 0: result += abs(y)`
which is potentially faster if `abs` is implemented via https://graphics.stanford.edu/~seander/bithacks.html#IntegerAbs (haven't benchmarked nor looked at generated asm)

Reply:
Will consider it in the future